samples/embedded-auth-with-sdk/src/main/java/com/okta/spring/example/controllers/LoginController.java

```diff
@@ -134,8 +145,67 @@ public ModelAndView selectAuthenticator(final @RequestParam("authenticator-type"
+                for (Authenticator authenticator : authenticators) {
+                    if (authenticator.getLabel().equals("Security Key or Biometric")) {
+                        authId = authenticator.getId();
+                    }
+                }
+
+                AuthenticationResponse enrollResponse = idxAuthenticationWrapper.enrollAuthenticator(proceedContext, authId);
+                idxResponse = idxAuthenticationWrapper.getClient().
+                        introspect(enrollResponse.getProceedContext().getClientContext());
+                //logger.info("IDX Response {}", idxResponse.raw());
+                authenticators = enrollResponse.getAuthenticators();
+                Util.updateSession(session, enrollResponse.getProceedContext());
+
+                RemediationOption[] remediationOptions = idxResponse.remediation().remediationOptions();
+                Optional<RemediationOption> remediationOptionsOptional = Arrays.stream(remediationOptions)
+                        .filter(x -> "select-authenticator-authenticate".equals(x.getName()))
+                        .findAny();
+
+                if (remediationOptionsOptional.isPresent()) {
+                    ModelAndView modelAndView = new ModelAndView("select-webauthn-authenticator");
+                    modelAndView.addObject("title", "Select Webauthn Authenticator");
+
+                    AuthenticatorEnrollments authenticatorEnrollments = idxResponse.getAuthenticatorEnrollments();
+                    Optional<AuthenticatorEnrollment> authenticatorEnrollmentOptional = Arrays.stream(authenticatorEnrollments.getValue())
+                            .filter(x -> "security_key".equals(x.getType()))
+                            .findAny();
+
+                    String webauthnCredentialId = authenticatorEnrollmentOptional.get().getCredentialId();
+                    ChallengeData challengeData = idxResponse.getCurrentAuthenticator().getValue().getContextualData().getChallengeData();
+                    modelAndView.addObject("webauthnCredentialId", webauthnCredentialId);
+                    modelAndView.addObject("challengeData", challengeData);
+                    return modelAndView;
+                }
+
+                remediationOptionsOptional = Arrays.stream(remediationOptions)
+                        .filter(x -> "select-authenticator-enroll".equals(x.getName()))
+                        .findAny();
+
+                if (remediationOptionsOptional.isPresent()) {
+                    ModelAndView modelAndView = new ModelAndView("enroll-webauthn-authenticator");
+                    modelAndView.addObject("title", "Enroll Webauthn Authenticator");
+                    modelAndView.addObject("currentAuthenticator", idxResponse.getCurrentAuthenticator());
+                    modelAndView.addObject("stateHandle", idxResponse.getStateHandle());
+                    modelAndView.addObject("href", idxResponse.remediation().getValue()[0].getHref());
```

Comment:
You could use the optional here instead too, I think.

Reply:
yep, will refine this.